src/explorer.api/Explorers/Exploration.cs

```diff
@@ -31,18 +32,69 @@ public Exploration(IEnumerable<ExplorerBase> explorers)
         public Task Completion { get; }
 
         public ExploreResult.ExploreStatus Status =>
-                Completion.Status switch
+            Completion.Status switch
+            {
+                TaskStatus.Canceled => ExploreResult.ExploreStatus.Canceled,
+                TaskStatus.Created => ExploreResult.ExploreStatus.New,
+                TaskStatus.Faulted => ExploreResult.ExploreStatus.Error,
+                TaskStatus.RanToCompletion => ExploreResult.ExploreStatus.Complete,
+                TaskStatus.Running => ExploreResult.ExploreStatus.Processing,
+                TaskStatus.WaitingForActivation => ExploreResult.ExploreStatus.Processing,
+                TaskStatus.WaitingToRun => ExploreResult.ExploreStatus.Processing,
+                TaskStatus.WaitingForChildrenToComplete => ExploreResult.ExploreStatus.Processing,
+                _ => throw new System.Exception("Unexpected TaskStatus: '{status}'."),
+            };
+
+        public static Exploration? Create(
+            AircloakQueryResolver resolver,
+            AircloakType columnType,
+            string tableName,
+            string columnName)
+        {
+            var components = columnType switch
+            {
+                AircloakType.Integer => new ExplorerBase[]
+                {
+                    new IntegerColumnExplorer(resolver, tableName, columnName, string.Empty),
+                    new MinMaxExplorer(resolver, tableName, columnName),
+                },
+                AircloakType.Real => new ExplorerBase[]
+                {
+                    new RealColumnExplorer(resolver, tableName, columnName),
+                    new MinMaxExplorer(resolver, tableName, columnName),
+                },
+                AircloakType.Text => new ExplorerBase[]
+                {
+                    new TextColumnExplorer(resolver, tableName, columnName),
+                    new EmailColumnExplorer(resolver, tableName, columnName),
+                    new IntegerColumnExplorer(resolver, tableName, $"length({columnName})", "text.length"),
```

Comment:
This is clever, I like it. I hadn't thought about the possibility of using an expression instead of a raw column name. 
👍